### PR TITLE
Add common code to check if services are enabled

### PR DESCRIPTION
**Pull request by pandeyabs**
Currently corso CLI & SDK have different checks for services enabled. These checks have diverged which can lead to bugs. For e.g. SDK users use [code](https://github.com/alcionai/corso/blob/3e43028a883bb2a38b3fe96eaeab745f31193445/src/pkg/services/m365/users.go#L92) to check if OD is enabled, while corso uses [this code](https://github.com/alcionai/corso/blob/main/src/pkg/services/m365/api/users.go#L174). These funcs have different checks. This PR introduces common helpers to consolidate these checks in one place.

- Note: I decided against absorbing these helpers into api/users.go, because separation of concerns -  getters shouldn't arbitrate on return vals. 
- Note that this code is not yet wired up to SDK/CLI. It'll be added in a following [PR](https://github.com/alcionai/corso/pull/4096). 

**Changes**
1. `Is*ServiceEnabled` helpers in internal/m365/common.go. 

2. Add `GetMailboxInfo` to common code. This is currently a duplicate of code in `GetInfo`. That function will be removed once we port SDK users to `GetMailboxInfo`. 

3. Unit tests for common code. Integration tests will be added in a later PR
- Note:`TestIsOneDriveServiceEnabled` is copied from `TestCheckUserHasDrives`. The older test will be removed in a later PR.

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [x] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* https://github.com/alcionai/corso/issues/3844

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/common.go
Comment (ryanfkeepers) [inline src/internal/m365/common.go:1]: Nit/soapbox.  I'm not going to block the PR on this, but I do think it's in our best interest to solve the issues described below.

Files named `common` are generally considered a smell by the golang community, as they indicate that the code either doesn't have a distinct enough identity to warrant separate ownership or that the identity is misunderstood, and the result often turns into a sort of dumping ground for code that could often be refined towards an identity.

In this case I think the file misunderstands the code's identity.  In one perspective it'd be a stronger identity just to rename the file "service_enablement.go".  But I'd still call that a smell, because it's suggesting that in addition to our current design where we package either by service (ex: m365/services/<service>)- or, if that's problematic, by purpose (ex: m365/collections/<collection type>)- that we also want to include files that have arbitrary combinations of service-specific code.  I think that'll make the choice of implementation location, and the discovery of code after implementation, more difficult in the long run.

Since each of these funcs are already clearly owned by one of our existing services, I think the cleanest approach is to split them into their respective service packages.  ie: `m365/services/*/enabled.go` (or whatever you want the filename to be).  That'll ensure that 1/ the code is easy to discover, 2/ there's no confusion about which code goes where, and 3/ future implementations with new services have a clear pattern to follow.

**Review by ryanfkeepers — APPROVED** [review]
Approving to unblock, but would really like to see the code movement as proposed by the big comment I left.  Otherwise the behavior LGTM.

File: src/internal/m365/common.go
Comment (pandeyabs) [reply src/internal/m365/common.go]: removed

**Pull request merged**
